### PR TITLE
Update docs for new CSS modules features

### PR DESCRIPTION
**Pull request by devongovett**
Docs for https://github.com/parcel-bundler/parcel-css/pull/180, https://github.com/parcel-bundler/parcel-css/pull/183, and https://github.com/parcel-bundler/parcel/pull/8122

### CODE REVIEW

File: src/languages/css.md [hunk 1]
```diff
@@ -141,9 +143,130 @@ import * as styles from './styles.module.css';
...
+```css
+:root {
+  --accent-color: hotpink;
+}
+```
+
+{% endsamplefile %}
+{% endsample %}
+
+To reference a global variable, use the `from global` syntax:
```
Comment (mischnic) [inline src/languages/css.md]: Is there a way to declare a global variable in a CSS module asset?

File: src/languages/css.md [hunk 2]
```diff
@@ -141,9 +143,130 @@ import * as styles from './styles.module.css';
...
+
+The following placeholders are currently supported:
+
+* `[name]` - The base name of the file, without the extension.
+* `[hash]` - A hash of the full file path.
+* `[local]` - The original class name or identifier.
+
+{% warning %}
+
+**Note**: CSS grid line names can be ambiguous due to automatic postfixing done by the browser, which generates line names ending with `-start` and `-end` for each grid template area. When using CSS grid, ensure that your `"pattern"` configuration ends with the `[local]` placeholder so that these references work correctly.
```
Comment (mischnic) [inline src/languages/css.md]: Nit: Doesn't parcel-css throw an error if the user didn't respect this? Then I'd rather phrase this as "When using CSS, your pattern configuation has to end ... "

File: src/languages/css.md [hunk 3]
```diff
@@ -141,9 +143,130 @@ import * as styles from './styles.module.css';
...
+
+{% endsamplefile %}
+{% endsample %}
+
+{% endwarning %}
+
 ### Enabling CSS modules globally
 
-By default, CSS modules are only enabled for files ending with `.module.css`. All other CSS files are treated as global CSS by default. However, this can be overridden to enable CSS modules for all files by configuring `@parcel/transformer-css` in your project root `package.json`.
+By default, CSS modules are only enabled for filenames ending with `.module.css`. All other CSS files are treated as global CSS by default. However, this can be overridden to enable CSS modules for all source files (i.e. not in `node_modules`) by configuring `@parcel/transformer-css` in your project root `package.json`.
```
Comment (mischnic) [inline src/languages/css.md]: How can CSS modules be enabled for a filename?
Maybe "for files whose name ends with"?

File: src/languages/css.md (hunk 1 above)
Comment (devongovett) [reply src/languages/css.md]: no, you'd have to do it by importing a non-css module. I can note this

**Pull request merged**